configargparse.py

```diff
@@ -223,12 +223,23 @@ def parse(self, stream):
                 "yaml.load('%s') returned type '%s' instead of 'dict'." % (
                 getattr(stream, 'name', 'stream'),  type(parsed_obj).__name__))
 
+        def _kvgen(d, pfix=None):
+            for k, v in d.items():
+                if pfix:
+                    npfix = pfix + '-' + k
+                else:
+                    npfix = k
+
+                if isinstance(v, dict):
+                    yield from _kvgen(v, npfix)
```

Comment:
This is cool syntax, but not supported by python2.7.  
I'd like to maintain python2.7 support for a while longer.